fem/pbilinearform.cpp

```diff
@@ -295,6 +295,39 @@ const
    pfes->Dof_TrueDof_Matrix()->MultTranspose(a, Y, 1.0, y);
 }
 
+double ParBilinearForm::InnerProduct(const ParGridFunction &x,
+                                     const ParGridFunction &y)
+{
+   MFEM_ASSERT(x.ParFESpace() == pfes, "the parallel spaces must agree");
+   MFEM_ASSERT(y.ParFESpace() == pfes, "the parallel spaces must agree");
+
+   HypreParVector *x_p = x.ParallelProject();
+   HypreParVector *y_p = y.ParallelProject();
+
+   double res = InnerProduct(*x_p, *y_p);
```

Comment:
Something is probably missing here. The intent is to do `x^t A y` right? I don't see the action of the `ParBilinearForm`.

Reply:
It calls the new `ParBilinearForm::InnerProduct()` with `HypreParVector` parameters, I believe :wink:

Reply:
> It calls the new `ParBilinearForm::InnerProduct()` with `HypreParVector` parameters, I believe 😉

Yes ... I missed that. Thanks !